<?xml version="1.0"?>
<!--USAGE
full demo:
roslaunch elevator elevator.launch gazebo:=true x:=1 y:=-2

rosrun rviz rviz -d `rospack find armadillo2_navigation`/rviz/amcl.rviz

only_elevator:
roslaunch elevator elevator.launch gazebo:=true Y:=0.3

rosrun rviz rviz -d `rospack find armadillo2_navigation`/rviz/amcl.rviz

roslaunch armadillo2 armadillo2.launch moveit:=true intel_cam:=true lidar:=true kinect:=true
-->

<launch>


 <!--settings for elevator-->
 <arg name="button_img" default="$(find elevator)/img/button_real.png" doc="path to button image"/>
 <arg name="pressed_button_img" default="$(find elevator)/img/button_sim_pressed.png" doc="path to pressed button image"/>
 <arg name="inner_button_img" default="$(find elevator)/img/level2_sim.png" doc="path to inner button image"/>
 <arg name="pressed_inner_button_img" default="$(find elevator)/img/level2_pressed_sim.png" doc="path to pressed inner button image"/>
 <arg name="press_offset_x" default="0.75" doc="offset from button_img to left corner to pushing spot"/>
 <arg name="press_offset_y" default="0.0" doc="offset from button_img to left corner to pushing spot"/>

  <!-- <node name="arm_server_node" pkg="elevator" type="arm_server_node.py" output="screen"/> -->

   <node name="armadillo_elevator_node" pkg="elevator" type="detect_button.py" output="screen">
     <param name="button_img" value="$(arg button_img)"/>
     <param name="pressed_button_img" value="$(arg pressed_button_img)"/>
     <param name="inner_button_img" value="$(arg inner_button_img)"/>
     <param name="pressed_inner_button_img" value="$(arg pressed_inner_button_img)"/>
     <param name="press_offset_x" value="$(arg press_offset_x)"/>
     <param name="press_offset_y" value="$(arg press_offset_y)"/>
   </node>

</launch>
